### PR TITLE
Feature/product list/filter component

### PR DESCRIPTION
**Pull request by KazeUmiAI**
# Description

Describe what you have done in this PR.

# Checklist

- [ ] This is a breaking changes?

# Screenshots

Provide screenshots or videos of the changes made if any.

Please provide screenshots or videos:

- From `playground` if you making changes in `@namviet-fe/core-ui`
- From `next js app` if your making any changes in `@namviet-fe/next-app`.

### CODE REVIEW

File: packages/core-components/src/components/product-list/FilterMobile.tsx [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+            viewBox='0 0 10 6'
+            fill='none'
+            xmlns='http://www.w3.org/2000/svg'
+          >
+            <path d='M0 0.5L5 5.5L10 0.5H0Z' fill='#575757' />
+          </svg>
+        </button>
+        <div className='flex divide-x'>
+          <button className='px-3' onClick={changePageLayout}>
+            <svg
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterMobile.tsx:78]: Can we move this somewhere else to make this comp easier to read a?

File: packages/core-components/src/components/product-list/FilterItem.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  const [isOpen, setIsOpen] = useState(false);
+  return filterItems?.filterType && filterItems?.filterOptions ? (
+    <div className='font-primary'>
+      <div className='text-lg/5 font-light'>
+        <p className='font-semibold' style={{ color: '#850000' }}>
+          {filterItems.title}
+        </p>
+        <div className='h-[70px]'>
+          <button
+            id='dropdownButton'
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterItem.tsx]: what's the purpose of this `id`,a ?

File: packages/core-components/src/components/product-list/FilterItem.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+const FilterItem: React.FC<FilterItemProps> = ({
+  filterItems,
+  onClickSort,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  return filterItems?.filterType && filterItems?.filterOptions ? (
+    <div className='font-primary'>
+      <div className='text-lg/5 font-light'>
+        <p className='font-semibold' style={{ color: '#850000' }}>
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterItem.tsx]: Please use color-primary

File: packages/core-components/src/components/product-list/FilterItem.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+            }}
+          >
+            <p
+              className='line-clamp-2 font-normal text-sm/4'
+              style={{ color: '#353535' }}
+            >
+              {filterItems.filterOptions.find((item) => item.selected)
+                ?.optionName ?? filterItems.placeholder}
+            </p>
+            <svg
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterItem.tsx:52]: can we move this one somewhere else

File: packages/core-components/src/components/product-list/FilterItem.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                  strokeLinejoin='round'
+                  strokeWidth={2}
+                  d='M13 7 7.674 1.3a.91.91 0 0 0-1.348 0L1 7'
+                />
+              )}
+            </svg>
+          </button>
+          {isOpen ? (
+            <div
+              id='dropdown'
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterItem.tsx]: what's the purpose of this id,a ?

Comment (its-rav) [reply packages/core-components/src/components/product-list/FilterItem.tsx]: Could we split this into a separate component, a? make it easier to read

File: packages/core-components/src/components/product-list/FilterItem.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+                className='divide-y divide-[#7D7D7DE5] p-1'
+                style={{
+                  boxShadow: '0 2px 6px 0 rgba(0, 0, 0, 0.15)',
+                  color: 'rgba(125, 125, 125, 0.9)',
+                }}
+              >
+                {filterItems.filterOptions.map((item) => {
+                  return item.selected ? (
+                    <li key={item.optionId}>
+                      <p className='block p-4' style={{ color: '#850000' }}>
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterItem.tsx]: color-primary

File: packages/core-components/src/components/product-list/FilterMobile.tsx [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+              );
+            })}
+          </ul>
+        </div>
+      ) : (
+        <></>
+      )}
+      {/* List Filter */}
+      {isOpenFilter ? (
+        <div className='fixed inset-0 bg-black bg-opacity-30 backdrop-blur-sm'>
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterMobile.tsx:149]: can we split this out a

File: packages/core-components/src/components/product-list/FilterItem.tsx (hunk 2 above)
Comment (KazegamiKuon) [reply packages/core-components/src/components/product-list/FilterItem.tsx]: aria-labelledby='dropdownButton'

File: packages/core-components/src/components/product-list/FilterItem.tsx (hunk 3 above)
Comment (KazegamiKuon) [reply packages/core-components/src/components/product-list/FilterItem.tsx]: updated

File: packages/core-components/src/components/product-list/FilterItem.tsx (hunk 4 above)
Comment (KazegamiKuon) [reply packages/core-components/src/components/product-list/FilterItem.tsx:52]: can move to config or icon folder but i won't change it so you can move it by your self

File: packages/core-components/src/components/product-list/FilterItem.tsx (hunk 5 above)
Comment (KazegamiKuon) [reply packages/core-components/src/components/product-list/FilterItem.tsx]: data-dropdown-toggle='dropdown'

File: packages/core-components/src/components/product-list/FilterItem.tsx (hunk 6 above)
Comment (KazegamiKuon) [reply packages/core-components/src/components/product-list/FilterItem.tsx]: updated

File: packages/core-components/src/components/product-list/FilterItem.tsx (hunk 5 above)
Comment (its-rav) [reply packages/core-components/src/components/product-list/FilterItem.tsx]: then should we add some prefix for it, a? product-list-dropdown?

File: packages/core-components/src/components/product-list/FilterItem.tsx (hunk 4 above)
Comment (KazeUmiAI) [reply packages/core-components/src/components/product-list/FilterItem.tsx:52]: We can but didn't neccesarily

File: packages/core-components/src/components/product-list/FilterMobile.tsx (hunk 7 above)
Comment (KazeUmiAI) [reply packages/core-components/src/components/product-list/FilterMobile.tsx:149]: we can but it take efforts

File: packages/core-components/src/components/product-list/FilterItem.tsx (hunk 5 above)
Comment (KazeUmiAI) [reply packages/core-components/src/components/product-list/FilterItem.tsx]: Updated, I forgot something

File: packages/core-components/src/components/product-list/FilterItem.tsx [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+export type FilterItemType = {
+  placeholder: string;
+  title: string;
+  filterId: string;
+  filterType: string;
+  filterOptions: OptionType[];
+};
+type FilterItemProps = {
+  filterItems: FilterItemType;
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterItem.tsx:17]: typo

File: packages/core-components/src/components/product-list/FilterItem.tsx [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    <div className='font-primary'>
+      <div className='text-lg/5 font-light'>
+        <p className='font-semibold color-primary'>{filterItems.title}</p>
+        <div className='h-[70px]'>
+          <button
+            id={`dropdown-button-${filterItems.filterType}`}
+            data-dropdown-toggle={`dropdown-${filterItems.filterType}`}
+            data-dropdown-delay='500'
+            className='w-full p-4 rounded-md border-solid flex justify-end items-center mt-4 hover:bg-gray-200'
+            style={{
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterItem.tsx:36]: theme color

File: packages/core-components/src/components/product-list/FilterMobile.tsx [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+      filterData.push(filterValue);
+      setFilterData([...filterData]);
+    }
+  };
+  const changePageLayout = () => {
+    setListLayout(!isListLayout);
+    onChangePageLayout?.(!isListLayout);
+  };
+  return (
+    <div className=''>
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterMobile.tsx:62]: redundant

File: packages/core-components/src/components/product-list/FilterMobile.tsx [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+      <div className='flex justify-between py-5 font-medium text-sm/4 border-b'>
+        <button className='flex items-center px-3' onClick={checkSort}>
+          <p className='pr-4'>{filterMobileItems.sortItems.title}</p>
+          <svg
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterMobile.tsx:66]: Can we move this somewhere else to make this comp easier to read a?

File: packages/core-components/src/components/product-list/FilterMobile.tsx [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+                d='M0 0.5V7.16667H6.66667V0.5H0ZM5 5.5H1.66667V2.16667H5V5.5ZM0 8.83333V15.5H6.66667V8.83333H0ZM5 13.8333H1.66667V10.5H5V13.8333ZM8.33333 0.5V7.16667H15V0.5H8.33333ZM13.3333 5.5H10V2.16667H13.3333V5.5ZM8.33333 8.83333V15.5H15V8.83333H8.33333ZM13.3333 13.8333H10V10.5H13.3333V13.8333Z'
+                fill='#575757'
+              />
+            </svg>
+          </button>
+          <button
+            className='flex items-center px-3'
+            onClick={() => {
+              setIsOpenFilter(true);
+              // setIsOpenSort(false);
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterMobile.tsx:97]: unuse comment

File: packages/core-components/src/components/product-list/FilterMobile.tsx [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+            }}
+          >
+            <p className='pr-5'>{filterTitle}</p>
+            <svg
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterMobile.tsx:101]: Can we move this somewhere else to make this comp easier to read a?

File: packages/core-components/src/components/product-list/FilterMobile.tsx [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+                  stroke='currentColor'
+                  strokeLinecap='round'
+                  strokeLinejoin='round'
+                  strokeWidth='2'
+                  d='m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6'
+                />
+              </svg>
+            </button>
+            {filterMobileItems?.filterItems ? (
+              <>
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterMobile.tsx:174]: redundant

File: packages/core-components/src/components/product-list/FilterMobile.tsx [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+                </li>
+              );
+            })}
+          </ul>
+        </div>
+      ) : (
+        <></>
+      )}
+      {/* List Filter */}
+      {isOpenFilter ? (
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterMobile.tsx:148]: too many render condition here, a? reason why we need to split into smaller components

File: apps/next-app/src/app/products/page.tsx [hunk 16]
```diff
@@ -77,16 +183,25 @@ export default function ProductsPage() {
...
               {title.toUpperCase()}
             </span>
             <span className='text-[#575757] md:text-[#000000] text-[11px] md:text-base font-normal'>{`(${quantity} sản phẩm)`}</span>
           </div>
-          <div className='col-span-12 md:col-span-3'></div>
-          <div className='col-span-12 md:col-span-9'>
+          <div className='col-span-12 md:!col-span-3'>
+            <FilterComponent
+              filterGroupItems={filterData}
+              applyFilterTitle={'Áp dụng'}
```
Comment (its-rav) [inline apps/next-app/src/app/products/page.tsx:197]: translation

File: packages/core-components/src/components/product-list/FilterMobile.tsx [hunk 17]
```diff
@@ -0,0 +1,246 @@
...
+                />
+              </svg>
+            </button>
+            {filterMobileItems?.filterItems ? (
+              <>
+                {filterMobileItems.filterItems.map((filterItem) => {
+                  return (
+                    <div
+                      key={filterItem.filterId}
+                      className='py-6 grid-cols-1 '
```
Comment (its-rav) [inline packages/core-components/src/components/product-list/FilterMobile.tsx:179]:
```suggestion
                      className='py-6 grid-cols-1'
```

**Review by its-rav — APPROVED** [review]

**Pull request merged**